doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,394 @@
+to ensure that SymPy solves for the desired variable. 
+To ensure SymPy will produce results in a consistent format, use `dict=True`. 
+To extract the solutions, you can iterate through the list of dictionaries:  
+    
+```py
+>>> from sympy import parse_expr, solve
+>>> from sympy.abc import x
+>>> from sympy.parsing.sympy_parser import convert_equals_signs, standard_transformations
+>>> expr = "x ** 2 = y"
+>>> parsed = parse_expr(expr, transformations=standard_transformations + (convert_equals_signs,))
+>>> print(parsed)
+Eq(x**2, y)
+>>> solutions = solve(parsed, x, dict=True)
+>>> print(solutions)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+>>> for solution in solutions:
+...     for key, val in solution.items():
+...         print(val)
+-sqrt(y)
+sqrt(y)
+```
+
+If you already have the equation in `Eq` form, you can parse that string:
+
+```py
+>>> from sympy import parse_expr, solve
+>>> from sympy.abc import x
+>>> expr = "Eq(x**2, y)"
+>>> parsed = parse_expr(expr)
+>>> print(parsed)
+Eq(x**2, y)
+>>> solutions = solve(parsed, x, dict=True)
+>>> print(solutions)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+>>> for solution in solutions:
+...     for key, val in solution.items():
+...         print(val)
+-sqrt(y)
+sqrt(y)
+```
+
+### Restricting the domain of solutions using {func}`~.solve`
+
+By default, SymPy will return solutions in the complex domain, which also includes 
+purely real and imaginary values. Here, the first two solutions are real, 
+and the last two are imaginary:
+
+```py
+>>> from sympy import Symbol, solve
+>>> x = Symbol('x')
+>>> solution = solve(x ** 4 - 256, x)
+>>> print(solution)
+[-4, 4, -4*I, 4*I]
+```
+
+If you want to restrict returned solutions to real numbers, you can place an assumption on the 
+symbol to be solved for, $x$:
+
+```py
+>>> from sympy import Symbol, solve
+>>> x = Symbol('x', real=True)
+>>> solution = solve(x ** 4 - 256, x)
+>>> print(solution)
+[-4, 4]
+```
+
+### Substituting solutions from {func}`~.solve` into an expression
+
+You can substitute solutions from {func}`~.solve` into an expression.
+
+A common use case is finding the critical points and values for a function $f$. 
+At the critical points, the derivative equals zero (or is undefined). 
+You can then obtain the function values at those critical points
+by substituting the critical points back into the function using {meth}`~sympy.core.basic.Basic.subs`.
+
+```py
+>>> from sympy.abc import x
+>>> from sympy import solve, diff
+>>> f = x**3 + x**2 - x
+>>> derivative = diff(f, x)
+>>> critical_points = solve(derivative, x, dict=True)
+>>> print(critical_points)
+[{x: -1}, {x: 1/3}]
+>>> for critical_point in critical_points:
+...     # Extract the value from the dictionary where key is x
+...     critical_point_value = critical_point[x]
+...     f_at_critical_point = f.subs(x, critical_point_value)
+...     print(f"f({critical_point_value}) = {f_at_critical_point}")
+f(-1) = 1
+f(1/3) = -5/27
```

Comment:
This is more complicated than it needs to be. I would just show this as:
```python
In [5]: critical_points = solve(derivative, x, dict=True)

In [6]: critical_points
Out[6]: [{x: -1}, {x: 1/3}]

In [7]: point1, point2 = critical_points

In [8]: f.subs(point1)
Out[8]: 1

In [9]: f.subs(point2)
Out[9]: -5/27
```
The point is that a dict can be used directly with `subs`. That's why a list of dicts is the best output because the main purpose of `solve` is that you will substitute the solution into something.